### ISSUE TITLE
bug with multi-dimension arrays

### ISSUE THREAD

**Issue by ninegua** [IntelLabs/numba#2]
The program below give wrong output (and crash):

```
import numba
import numpy as np
@numba.njit(parallel=True)
def ff(a, b):
    return a * b
a = np.arange(8).reshape(8,1) + 1.0 
b = np.arange(8).reshape(8,1) + 1.0 
print(ff(a,b)) 
```

**Comment by ehsantn** [comment]
It works with sequential lowering so it's probably something gufunc backend related.

**Comment by ehsantn** [comment]
@DrTodd13 this might be related to the issues you are looking at.

**Comment by ninegua** [comment]
The program would work for `3x3`, or `1x8`. So I tried reversing the order of `loop_indices` when forming the nested loop based on `sched` in `_create_gufunc_for_parfor_body`, then it all works out. Likewise, we could change the order of indices in `sched` to fix it. So there is a mis-match between inner-outer loop indices assumed by parfor's loop nests and those arranged in `sched` array.

What is the order of loop nests? I'd assume inner dimension is the inner loop nest. Just want to confirm if that is the case.

**Comment by ninegua** [comment]
@DrTodd13 You might want to take care of this. 

If `loop_indices` goes by outer dimension to inner dimension, then the loop nest would already have been the correct order, and reversing `loop_indices` would be wrong. So instead we should read (or produce) `sched` differently.

**Comment by DrTodd13** [comment]
I think it is correct now with 67c23ce..2b0fd63.

**Comment by DrTodd13** [comment]
I'll be doing more multi-D parfor testing.

**Comment by ninegua** [comment]
It turns out the real problem was because symbols in gu_signature was given wrong dimensional data, which in turn messed up multi-dimensional array indexing. Although the exhibited symptom seemed to be with wrong order of indices, my analysis above was wrong and didn't uncover the root cause.

Now this bug is fixed by 6ae107e016f4169f7c1d123410ca2c0d66b7bc1b, and proper order of indexing `sched` is also restored.
